libraries/AP_DroneCAN/AP_DroneCAN_DNA_Server.cpp

```diff
@@ -112,12 +113,9 @@ bool AP_DroneCAN_DNA_Server::Database::handle_node_info(uint8_t source_node_id,
     WITH_SEMAPHORE(sem);
 
     if (is_registered(source_node_id)) {
-        // verify that the unique ID matches our registration for the node ID
-        if (source_node_id == find_node_id(unique_id, 16)) {
-            return false;
-        } else {
-            // this device's node ID is associated with a different device
-            return true;
+        // this device's node ID is associated with a different unique ID
+        if (source_node_id != find_node_id(unique_id, 16)) {
+            return true; // so raise as duplicate
         }
     } else {
```

Comment:
I'd be tempted to kill this else clause and return false in two places, one for the registered, one for the unregister.

Reply:
Thanks for the input, I’ll look at this in the next PRs, still got lots more changes to go: